mesonbuild/interpreter.py

```diff
@@ -3169,9 +3169,28 @@ def func_project(self, node, args, kwargs):
         if not self.is_subproject():
             self.build.project_name = proj_name
         self.active_projectname = proj_name
-        self.project_version = kwargs.get('version', 'undefined')
-        if not isinstance(self.project_version, str):
-            raise InvalidCode('The version keyword argument must be a string.')
+        version = kwargs.get('version', 'undefined')
+        if isinstance(version, list):
+            if len(version) != 1:
+                raise InvalidCode('Version argument is an array with more than one entry.')
+            version = version[0]
+        if isinstance(version, mesonlib.File):
```

Comment:
Missing FeatureNew()

Reply:
Added. This is the first one of these that I have added, hopefully I got it right.

Reply:
looks good.